.github/workflows/enzyme-rust.yml

```diff
@@ -0,0 +1,62 @@
+name: Enzyme Rust CI
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  merge_group:
+
+jobs:
+  rust-autodiff:
+    name: Rust Autodiff Tests LLVM ${{ matrix.llvm }}
+    runs-on: [self-hosted, Linux, X64, 32-core]
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        llvm: ["19", "20"]
+    
+    timeout-minutes: 90
+    
+    steps:
+    - name: Install LLVM and dependencies
+      run: |
+        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add -
+        sudo apt-add-repository "deb http://apt.llvm.org/`lsb_release -c | cut -f2`/ llvm-toolchain-`lsb_release -c | cut -f2`-${{ matrix.llvm }} main" || true
+        sudo apt-get install -y cmake gcc g++ llvm-${{ matrix.llvm }}-dev libomp-${{ matrix.llvm }}-dev lld-${{ matrix.llvm }} clang-${{ matrix.llvm }} libclang-${{ matrix.llvm }}-dev libeigen3-dev libboost-dev libzstd-dev
+        sudo python3 -m pip install --upgrade pip lit
+    - name: Install Rust toolchain
+      run: |
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+        source ~/.cargo/env
+        rustup toolchain install nightly
```

Comment:
until Rust adds CI for Enzyme, I think this should be fixed to a specific commit